syntax = "proto2";

package RpcMeta;

enum MessageType
{
  REQUEST = 1;
  RESPONSE = 2;
  ERROR = 3; 
}

enum ErrorCode
{
  NO_ERROR = 0;
  WRONG_PROTO = 1;
  NO_SERVICE = 2;
  NO_METHOD = 3;
  INVALID_REQUEST = 4;
  INVALID_RESPONSE = 5;
  TIMEOUT = 6;
}

message RpcMetaMessage
{
  required MessageType type = 1;
  required fixed64 id = 2;      //客户端每一次发起请求的消息id，服务端处理完成后再将id带回来

  optional string service = 3;
  optional string method = 4;
  optional bytes request = 5;

  optional bytes response = 6;

  optional ErrorCode error = 7;
}
